### PR TITLE
add `osx_arm64` to the build platforms

### PR DESCRIPTION
**Pull request by zacharyburnett**
I used this guide to add the build platform: https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: conda-forge.yml [hunk 1]
```diff
@@ -4,3 +4,7 @@ bot:
 github:
   branch_name: main
   tooling_branch_name: main
+
+build_platform:
+  osx_arm64: osx_64
+test_on_native_only: true
```
Comment (zacharyburnett) [inline conda-forge.yml:10]: added build platform here

**Comment by zacharyburnett** [comment]
It looks like the `arm64` build fails with the following errors:
```
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - cfitsio 4.1.0.*
 - libgfortran5 >=11.3.0
 - llvm-openmp 14.*
 - libgfortran 5.*
 - libgcc

with channels:

The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested libgcc
- 
```
```
conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform osx-arm64: {MatchSpec("libgcc")}
```
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=556408&view=logs&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a&t=f7bcb349-6dca-53ba-d88d-1d9963e119ba&l=133

**Comment by jhunkeler** [comment]
```diff
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 8a9cd41..7bc1bda 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -24,7 +24,7 @@ requirements:
     - libcurl  # [linux]
     - cfitsio
     - llvm-openmp  # [osx]
-    - libgcc
+    - libgcc  # [osx64]
     - _openmp_mutex  # [linux]
```
I was able to build it locally on a M1 by omitting `libgcc` on `arm64`.

**Review by jhunkeler — APPROVED** [review]
Looks good to me. I don't have anything else to add.

**Comment by zacharyburnett** [comment]
> Looks good to me. I don't have anything else to add.

Thanks! I don't have write access, could you merge this and https://github.com/conda-forge/fitsverify-feedstock/pull/4?

**Comment by jhunkeler** [comment]
:rocket:

**Pull request merged**